### ISSUE TITLE
High CPU load for systemd & fleetd when there are many units in loaded (but never started) state.

### ISSUE THREAD

**Issue by ewoutp** [coreos/fleet#1163]
When we `fleetctl load` a lot (500) of units onto a 3 machines AWS CoreOS cluster the cpu load of both systemd and fleetd goes up a lot.

Systemd uses about 30%, fleetd about 15% CPU time.

When we then start all those units (test units that only sleep forever) the systemd cpu usage drops to almost 0% and that of fleetd to around 11%.

If we then stop the units again, the cpu load remains about the same (and the units are still listed in `systemctl list-units` as expected)

My guess is that this is caused by fleet trying to get UnitProperties from every individual loaded unit (see https://github.com/coreos/fleet/blob/master/systemd/manager.go#L238).

Can you confirm that? If so what is the reason for having these individual getUnitState requests?

This has been tested on CoreOS version 522.4.0

**Comment by denderello** [comment]
We created the units and called the commands from within the cluster using [giantswarm/fleet-stress](https://github.com/giantswarm/fleet-stress).

**Comment by denderello** [comment]
I collected some numbers for this about the amount of request fleet is running against systemd. To get this information I ran `strace` and filtered for `write` syscalls on the file descriptor that fleet uses for systemd dbus like this: `strace -ff -e trace=write -s 256 -p $(pidof fleetd) 2>&1 |  stdbuf -o0 grep -E 'write\(4,'`

After submiting 500 units I could see 12 writes within a period of 10 seconds. After loading those units this number rose to 2016 writes (2004 of them where `GetAll` operations judging from the output).
